tools/ansible/tasks/wsu/main.yaml

```diff
@@ -73,16 +76,65 @@
+    # as OpenShift CI doesn't have the actual version attached to its clusters, instead replacing it with 0.0.1 and
+    # information about the release creation date
+    - name: Get cluster version
+      delegate_to: localhost
+      shell: |
+        MINORVERSION={{ kubernetes_version.stdout }}
+        BASEK8SVERSION=16
+        BASEOPENSHIFTVERSION=3
+        if [ \"$MINORVERSION\" -lt \"$BASEK8SVERSION\" ]; then
+        echo unsupported kubernetes version; exit 1
+        fi
+        VERSIONINCREMENTS=$((MINORVERSION - BASEK8SVERSION))
+        echo 4.$((BASEOPENSHIFTVERSION + VERSIONINCREMENTS))
+        exit 0
+      register: cluster_version
+
+    # curl -s https://api.github.com/repos/openshift/windows-machine-config-operator/releases gets the list of all
+    # releases, we use the cluster version from previous ansible task to filter out the release we need.
+    # The wmcb release tags are supposed to confirm to the following semver format.
+    # v<openshift-major-version>.<openshift-major-version>.<wmcb_build_number>-<release_type>. For example, v4.4-1-alpha.
+    # For a <openshift-major-version-version>.<openshift-minor-version>, there can be many wmcb builds available, for
+    # example, OpenShift 4.4 can map to v4.4.1-alpha, v4.4.2-alpha, we will always pick up the latest build available
+    # which is v4.4.2-alpha,
+    # In future, when we cut beta and GA releases, the playbook should always go for GA version when available.
+    - name: Get hybrid overlay url
+      delegate_to: localhost
+      shell: |
+        RELEASES="$(curl -s https://api.github.com/repos/openshift/windows-machine-config-operator/releases)"
+        TAGNAME="$(echo $RELEASES | jq '.[]' | jq '.tag_name' | grep -i {{ cluster_version.stdout }} | sort -r | awk 'NR==1')"
+        echo $RELEASES |
+        jq '.[] | select (.tag_name == '$TAGNAME')' | \
+        jq ' .assets[] | select (.name == "{{ hybrid_overlay_exe }}" )' | \
+        jq -r .browser_download_url
+      register: hybrid_overlay_download_url
+
     - name: Get hybrid overlay exe
       win_get_url:
-        url: "https://github.com/openshift/windows-machine-config-operator/releases/download/v0.3-alpha/hybrid-overlay.exe"
+        url: "{{ hybrid_overlay_download_url.stdout }}"
         dest: "{{ win_temp_dir.path }}\\hybrid-overlay.exe"
         follow_redirects: all
 
+    # We'll get the SHA associated with the hybrid_overlay.exe in the body of the release tag.
+    - name: Get the SHA for the given binary. We assume that body of WMCB release has the info.
+      delegate_to: localhost
+      shell: "curl -s https://api.github.com/repos/openshift/windows-machine-config-operator/releases | jq '.[] | select (.tag_name == '$(curl -s https://api.github.com/repos/openshift/windows-machine-config-operator/releases |  jq '.[]' | jq '.tag_name' | grep -i {{ cluster_version.stdout }} | sort -r | awk 'NR==1')')'| jq -r .body | grep -i {{ hybrid_overlay_exe }}| awk '{print $1}'"
```

Comment:
Sorry I did not clearly explain my previous comment. I was hoping you make the `curl -s https://api.github.com/repos/openshift/windows-machine-config-operator/releases` only once per playbook run i.e. curl it once and use that value everywhere in the playbook.

Reply:
You explained it properly I just missed this section of the code for whatever reason. Its been fixed